From 3cb3f2269dadc2d14b5a17dfbd0c52b4f425c9e0 Mon Sep 17 00:00:00 2001
From: Krisztian Litkey <krisztian.litkey@intel.com>
Date: Wed, 21 Dec 2016 14:46:03 +0200
Subject: [PATCH 1/1] build: allow controlling gobject introspection data
 generation.

Added --enable-gobject-introspection to allow forcing gobject
instrospection data generation off.

Signed-off-by: Krisztian Litkey <krisztian.litkey@intel.com>
---
 configure.ac | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 008728d..22614e3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -111,7 +111,13 @@ AM_CONDITIONAL(HAVE_LIBSOUP_CLIENT_CERTS, test x$have_libsoup_client_certs = xye
 m4_ifdef([GOBJECT_INTROSPECTION_CHECK], [
   GOBJECT_INTROSPECTION_CHECK([1.34.0])
 ])
-AM_CONDITIONAL(BUILDOPT_INTROSPECTION, test "x$found_introspection" = xyes)
+
+AC_ARG_ENABLE(gobject-introspection,
+              [AS_HELP_STRING([--enable-gobject-introspection],
+                              [generate gobject introspection data [default=auto]])],,
+              enable_gobject_introspection=$found_introspection)
+
+AM_CONDITIONAL(BUILDOPT_INTROSPECTION, test "x$found_introspection" = xyes -a "x$enable_gobject_introspection" != xno)
 
 LIBGPGME_DEPENDENCY="1.1.8"
 
@@ -346,4 +352,5 @@ AS_IF([test x$with_builtin_grub2_mkconfig = xyes], [
 ], [
     echo "    grub2-mkconfig path:                          $GRUB2_MKCONFIG"
 ])
+echo "    GObject Introspection data:                   $enable_gobject_introspection"
 echo ""
-- 
2.7.4

